app/src/main/java/com/github/onlynotesswent/ui/common/GridComposables.kt

```diff
@@ -76,30 +87,47 @@ fun CustomSeparatedLazyGrid(
                 FolderItem(
                     folder = folder,
                     navigationActions = navigationActions,
+                    isDeckView = isDeckView,
                     noteViewModel = noteViewModel,
+                    deckViewModel = deckViewModel,
                     folderViewModel = folderViewModel) {
                       folderViewModel.selectedParentFolderId(folder.parentFolderId)
                       navigationActions.navigateTo(
                           Screen.FOLDER_CONTENTS.replace(
                               oldValue = "{folderId}", newValue = folder.id))
                     }
               }
+              // Spacer item to create space between folders and notes
+              item(span = { GridItemSpan(maxLineSpan) }) {
+                Spacer(modifier = Modifier.height(50.dp))
+              }
             }
 
-            // Spacer item to create space between folders and notes
-            item(span = { GridItemSpan(maxLineSpan) }) { Spacer(modifier = Modifier.height(50.dp)) }
-
-            items(sortedNotes, key = { it.id }, span = { GridItemSpan(3) }) { note ->
-              NoteItem(
-                  note = note,
-                  currentUser = userViewModel.currentUser.collectAsState(),
-                  noteViewModel = noteViewModel,
-                  folderViewModel = folderViewModel,
-                  showDialog = false,
-                  navigationActions = navigationActions) {
-                    noteViewModel.selectedNote(note)
-                    navigationActions.navigateTo(Screen.EDIT_NOTE)
-                  }
+            if (isDeckView) {
+              items(sortedDecks, key = { it.id }, span = { GridItemSpan(3) }) { deck ->
+                DeckItem(
+                    deck = deck,
+                    deckViewModel = deckViewModel!!,
```

Comment:
The forced unwrapping of deckViewModel and noteViewModel is bad practice but in this case it is fine.

Reply:
Yes agreed its bad, I use `isDeckView` as a indicator to say if `deckViewModel` is not none or if `noteViewModel` is not none, but this requires the other screens that use this composable to correctly follow these rules. I won't be changing it as we unfortunately don't have enough time, but in all the screens that are shared between the note and deck screen I should first check if either `deckViewModel` or `noteViewModel` is not none before using them.